Assignment5/GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace GenericsHomework
+{
+    public class Node<T> : ICollection<T>
+    {
+        private T? _Data;
+
+        public T Data
+        {
+            get
+            {
+                return _Data!;
+            }
+            private set
+            {
+                _Data = value ??
+                    throw new ArgumentNullException(nameof(value));
+            }
+        }
+
+        private Node<T>? _Next;
```

Comment:
```suggestion
        private Node<T> _Next;
```
If you set this in the constructor it does not need to be nullable.